uuid/uuid.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2012 The gocql Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The uuid package can be used to generate and parse universally unique
+// identifiers, a standardized format in the form of a 128 bit number.
+//
+// http://tools.ietf.org/html/rfc4122
+package uuid
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+type UUID [16]byte
+
+var timeBase = time.Date(1582, time.October, 15, 0, 0, 0, 0, time.UTC).Unix()
+var hardwareAddr []byte
+var clockSeq uint32
+
+// TimeUUID generates a new time based UUID (version 1) using the current
+// time as the timestamp.
+func TimeUUID() UUID {
+	return UUIDFromTime(time.Now())
+}
+
+// UUIDFromTime generates a new time based UUID (version 1) as described in
+// RFC 4122. This UUID contains the MAC address of the node that generated
+// the UUID, the given timestamp and a sequence number.
+func UUIDFromTime(aTime time.Time) UUID {
+	var u UUID
+
+	utcTime := aTime.In(time.UTC)
+	t := uint64(utcTime.Unix()-timeBase)*10000000 + uint64(utcTime.Nanosecond()/100)
+	u[0], u[1], u[2], u[3] = byte(t>>24), byte(t>>16), byte(t>>8), byte(t)
+	u[4], u[5] = byte(t>>40), byte(t>>32)
+	u[6], u[7] = byte(t>>56)&0x0F, byte(t>>48)
+
+	clock := atomic.AddUint32(&clockSeq, 1)
```

Comment:
All this doesn't really matter. We just need a transaction ID that has an extremely high probably of being unique. Reading 8 bytes from `/dev/random` would do it, but it might be slow.